src/misceval.js

```diff
@@ -360,63 +339,65 @@ Sk.misceval.richCompareBool = function (v, w, op, canSuspend) {
         }
     }
 
-
     // handle identity and membership comparisons
     if (op === "Is") {
-        if (v instanceof Sk.builtin.int_ && w instanceof Sk.builtin.int_) {
-            return v.numberCompare(w) === 0;
-        } else if (v instanceof Sk.builtin.float_ && w instanceof Sk.builtin.float_) {
-            return v.numberCompare(w) === 0;
-        } else if (v instanceof Sk.builtin.lng && w instanceof Sk.builtin.lng) {
-            return v.longCompare(w) === 0;
+        if (v_type === w_type) {
+            if (v === w) {
+                return true;
+            } else if (v_type === Sk.builtin.float_) {
+                return v.v === w.v;
+            } else if (v_type === Sk.builtin.int_) {
+                return v.v === w.v;
+            } else if (v_type === Sk.builtin.lng) {
+                return v.longCompare(w) === 0;
+            }
         }
-
-        return v === w;
+        return false;
     }
 
     if (op === "IsNot") {
-        if (v instanceof Sk.builtin.int_ && w instanceof Sk.builtin.int_) {
-            return v.numberCompare(w) !== 0;
-        } else if (v instanceof Sk.builtin.float_ && w instanceof Sk.builtin.float_) {
-            return v.numberCompare(w) !== 0;
-        }else if (v instanceof Sk.builtin.lng && w instanceof Sk.builtin.lng) {
+        if (v_type !== w_type) {
+            return true;
+        } else if (v_type === Sk.builtin.float_) {
+            return v.v !== w.v;
+        } else if (v_type === Sk.builtin.int_) {
+            return v.v !== w.v;
+        } else if (v_type === Sk.builtin.lng) {
             return v.longCompare(w) !== 0;
         }
-
         return v !== w;
     }
 
     if (op === "In") {
         return Sk.misceval.chain(Sk.abstr.sequenceContains(w, v, canSuspend), Sk.misceval.isTrue);
     }
     if (op === "NotIn") {
-        return Sk.misceval.chain(
-            Sk.abstr.sequenceContains(w, v, canSuspend),
-            function(x) { return !Sk.misceval.isTrue(x); }
-        );
+        return Sk.misceval.chain(Sk.abstr.sequenceContains(w, v, canSuspend), function (x) {
+            return !Sk.misceval.isTrue(x);
+        });
```

Comment:
I am not a fan of mixing gratuitous, unnecessary formatting changes with real content changes. This pull request does a lot of that.

Reply:
Would you prefer a separate pr for formatting?

Reply:
The issue is that this makes it unnecessarily harder to use git blame when you need to figure out when/why something was done and it creates the potential for unnecessary conflicts when people have internal modifications (and even with other pending pull requests).  We have eslint set up for the project in an agreed upon way.  If eslint is okay with something, then I don't think it should be changed, especially when it's not relevant to the content of the PR.

If you don't like the agreed upon style, the community should have an independent conversation about that, and, in my opinion, if we change anything, the tools need to be set up to catch style violations.  That's what we did last time we cleaned up the code base.  Otherwise, new violations will be introduced and others will introduce "fixes" in subsequent pull requests.  It's a never ending rat hole for people to reformat code that is not to their liking.

Reply:
is this a deal breaker? 

I typically do a prettier format of the function I'm working on and so formatting changes here are just what prettier told me to do 😬 . 

But I get that it can take away from the content of the PR. And I try to make a judgement call about formatting going too far...

Reply:
how do we feel about const and let changes in this pr? is that the same or different to style changes?

Reply:
i've reverted some style changes and left the const changes.

Reply:
The formatting changes in this particular PR are not deal breakers for me, I am just making a point that I find these things to muddy the waters unnecessarily. Given that the current style is not based on what prettier does (whether or not it should be), I'm not a fan of changing things just because it says to do so.  Again, though, if we all decide we want to go that way, that's fine, but we should stop, merge all the pending pull requests that we are going to merge, and just reformat all the code in one go.  But, there are a bunch of pending pull requests out there, and every such formatting change could cause trouble for them.

I agree the const/let changes are actual improvements to the code, not stylistic preferences.